frontend/src/metabase/actions/components/ActionForm/ActionForm.unit.spec.tsx

```diff
@@ -0,0 +1,504 @@
+import React from "react";
+import _ from "underscore";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import { act } from "react-dom/test-utils";
+import type {
+  ActionFormProps,
+  ActionFormSettings,
+  FieldSettings,
+  WritebackParameter,
+} from "metabase-types/api";
+import type { Parameter } from "metabase-types/types/Parameter";
+
+import { ActionForm } from "./ActionForm";
+
+const makeFieldSettings = (
+  overrides: Partial<FieldSettings> = {},
+): FieldSettings => ({
+  id: "abc-123",
+  name: "form field name",
+  title: "form field name",
+  order: 1,
+  fieldType: "string",
+  inputType: "string",
+  required: false,
+  hidden: false,
+  ...overrides,
+});
+
+const makeParameter = (
+  overrides?: Partial<WritebackParameter>,
+): Partial<WritebackParameter> => {
+  const id = overrides?.id ?? "abc-123";
+  return {
+    id,
+    required: false,
+    target: ["variable", ["template-tag", id]],
+    type: "type/Text",
+    ...overrides,
+  };
+};
```

Comment:
Does it make sense to extract these to `metabase-types/api/mocks`?

Reply:
😬 I fear that making these general purpose would require a good bit more work and thinking, and would also make the tests more complex. I think (and I could be wrong about this), that these quick little mocks work best when they don't try to serve more than a few tests.